src/Piece/Queen.cpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include "Piece/Queen.h"
+
+bool Queen::isValidMove(const Position src, const Position des) const {
+    // Check if the move is along a straight line (horizontally, vertically, or diagonally)
+    if (src.row == des.row || src.col == des.col || abs(src.row - des.row) == abs(src.col - des.col)) {
+        // The move is along a straight line
+        return true;
+    }
```

Comment:
Use {} for if-else blocks: It's considered good practice to use braces {} for if-else blocks, even if they consist of only a single statement. This helps avoid potential bugs and improves code readability.